### PR TITLE
small changes

### PR DESCRIPTION
**Pull request by nxtdaydelivery**
## Description
- adding jetbrains annotation for ApiStatus to help prevent people use internal package etc
- string utils and its DSL
- comparable LocrawInfo and fix a NPE that happened once
- refractor test HUDs into their own pkg for docs

## Checklist
- [X] I made a clear description of what was changed
- [X] I stated why these changes were necessary
- [X] I updated documentation or said what needs to be updated
- [X] I made sure these changes are backwards compatible
- [X] This pull request is for one feature/bug fix

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -75,6 +76,7 @@ dependencies {
     compileOnly("org.apache.logging.log4j:log4j-api:2.0-beta9")
     compileOnly("org.ow2.asm:asm-debug-all:5.0.3")
     compileOnly("org.apache.commons:commons-lang3:3.3.2")
+    compileOnly("org.jetbrains:annotations:23.0.0")
```
Comment (Wyvest) [inline build.gradle.kts:79]: unless this is included in kotlin include annotations in the versions buildscript

File: src/main/java/cc/polyfrost/oneconfig/utils/hypixel/LocrawInfo.java [hunk 2]
```diff
@@ -111,6 +114,15 @@ public int hashCode() {
         return Objects.hash(serverId, gameMode, mapName, rawGameType);
     }
 
+    /**
+     * Compares the gametype of this locraw to another locraw.
+     * @see Comparable#compareTo(Object) 
+     */
+    @Override
+    public int compareTo(@NotNull LocrawInfo o) {
```
Comment (Wyvest) [inline src/main/java/cc/polyfrost/oneconfig/utils/hypixel/LocrawInfo.java:122]: how can this be compared to??????

**Review by Wyvest — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (nxtdaydelivery) [reply build.gradle.kts:79]: when I was doing it it said reference not resolved; try "add org.jetbrains:annotations to the classpath"

File: src/main/java/cc/polyfrost/oneconfig/utils/hypixel/LocrawInfo.java (hunk 2 above)
Comment (nxtdaydelivery) [reply src/main/java/cc/polyfrost/oneconfig/utils/hypixel/LocrawInfo.java:122]: I'll remove it tbh it's not very helpful

File: build.gradle.kts (hunk 1 above)
Comment (Wyvest) [reply build.gradle.kts:79]: > when I was doing it it said reference not resolved; try "add org.jetbrains:annotations to the classpath"

cool

**Review by Wyvest — CHANGES_REQUESTED** [review]

File: src/main/java/cc/polyfrost/oneconfig/utils/hypixel/HypixelUtils.java [hunk 3]
```diff
@@ -41,7 +41,7 @@
 import java.util.Locale;
 import java.util.concurrent.TimeUnit;
 
-/**
+    /**
```
Comment (xtrm-en) [inline src/main/java/cc/polyfrost/oneconfig/utils/hypixel/HypixelUtils.java:44]: why

Comment (nxtdaydelivery) [reply src/main/java/cc/polyfrost/oneconfig/utils/hypixel/HypixelUtils.java]: intellij moment

**Review by xtrm-en — DISMISSED** [review]
cool stuff :+1:

**Review by xtrm-en — DISMISSED** [review]

**Review by nxtdaydelivery — COMMENTED** [review]
lgtm (I can't approve as I am the author)

**Review by pauliesnug — DISMISSED** [review]
lgtm

**Review by xtrm-en — DISMISSED** [review]
lgtm

File: build.gradle.kts (hunk 1 above)
Comment (Wyvest) [inline build.gradle.kts:79]: coming back to this, is jetbrains annotations included in kotlin or not

because if it is then this will error

**Review by Wyvest — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (nxtdaydelivery) [reply build.gradle.kts:79]: it told me to include it.
it said unresolved reference
quick solution: add to classpath

**Pull request merged**